### ISSUE TITLE
Clarify toObservable VS outputToObservable replay behavior (in JSDoc)

### ISSUE THREAD

**Issue by arlowhite** [angular/angular#57224]
### Describe the problem that you experienced

A `Model` can be converted to an `Observable` with either `toObservable` or `outputToObservable`. The JSDocs do not make it clear that they have different replay behavior. Though this is described in https://angular.dev/guide/signals/rxjs-interop#timing-of-toobservable
> toObservable uses an effect to track the value of the signal in a ReplaySubject.

However, the `toObservable` JSDoc (docs you see in IDE) only say:
> The signal's value will be propagated into the Observable's subscribers using an effect.

### Describe what you were looking for in the documentation

Function JSdocs that help me decide whether I want `toObservable` or `outputToObservable`.

### Describe the actions that led you to experience the problem

I was using `switchMap`, which subscribes later, making replay a concern.

### Describe what you want to experience that would fix the problem

`toObservable` JSdocs should mention replay.

`outputToObservable` JSdocs should note that the Observable does not replay.

**Comment by devversion** [comment]
Agreed we should clarify this. Should be an easy addition to the JSDoc/written documentation. 

I think there are two key points:
- `outputToObservable` is creating an observable that represents the stream of "events firing" in an output.
- `toObservable` converts a signal that reflects state, hence subscribing via `ReplayObservable` makes sense as initial emit to reflect the "current state"
